File: ts/packages/core/src/ports/clock.ts
// Injected time source, never pulled from global Date.now() directly in domain logic -- lets token-expiry and revocation-timestamp checks be tested deterministically and lets a consumer substitute a synchronised network clock later without touching domain code.
export interface Clock {
  /** Current time as Unix milliseconds, matching every uint timestamp field in the protocol (token-claims.expires, revocation-entry.revoked-at, handshake params, etc.). */
  now: () => number;
}

File: rust/crates/wire-mesh-core/src/ports/clock.rs
//! The clock port. Every timestamp the protocol carries (token `expires`/`not-before`, `revoked-at`, handle `issued`/`expires`) is Unix milliseconds compared against this port, never against `SystemTime` directly, keeping expiry checks deterministic under test.

pub trait Clock: Send + Sync {
    /// Current Unix time in milliseconds.
    fn now_unix_ms(&self) -> u64;
}
